interface/modules/custom_modules/oe-module-gcip-auth/ModuleManagerListener.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/**
+ * GCIP Module Manager Listener
+ * 
+ * <!-- AI-Generated Content Start -->
+ * This class handles module lifecycle events such as installation,
+ * uninstallation, enabling, and disabling of the GCIP authentication
+ * module, ensuring proper database setup and cleanup.
+ * <!-- AI-Generated Content End -->
+ *
+ * @package   OpenEMR\Modules\GcipAuth
+ * @link      http://www.open-emr.org
+ * @author    OpenEMR Development Team
+ * @copyright Copyright (c) 2024 OpenEMR Development Team
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\GcipAuth;
+
+use OpenEMR\Events\ModuleManagerEvent;
+use OpenEMR\Modules\GcipAuth\Services\GcipAuditService;
+
+/**
+ * Module Manager Listener for GCIP Authentication
+ */
+class ModuleManagerListener
+{
```

Comment:
This won't work because Module Manager calls specific function names. See Weno module for an example.
Or you could improve how this class is called if wanted.